schema/trivia.schema.json

```
@@ -0,0 +1,54 @@
+{
+    "$id": "https://raw.githubusercontent.com/Cog-Creators/Red-DiscordBot/V3/develop/schema/trivia.schema.json",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Red-DiscordBot Trivia metadata file",
+    "type": "object",
+    "properties": {
+        "AUTHOR": {
+            "type": "string",
+            "description": "Author of the Trivia quiz."
+        },
+        "CONFIG": {
+            "type": "object",
+            "description": "The configuration for a trivia session.",
+            "required": [],
+            "properties": {
+                "bot_plays": {
+                    "type": "boolean",
+                    "description": "Whether or not the bot gains points during the session. Defaults to true.",
+                    "default": true
+                },
+                "delay": {
+                    "type": ["integer", "number"],
+                    "description": "Answer time limit in seconds, must be positive number greater than 0.0. Defaults to 15.0.",
+                    "default": 15.0
+                },
+                "max_score": {
+                    "type": "integer",
+                    "description": "Number of points required in order win the trivia, must be positive integer. Defaults to 10.",
+                    "default": 10
+                },
+                "payout_multiplier": {
+                    "type": ["integer", "number"],
+                    "description": "The payout multiplier, must be positive number. Defaults to 0.0.",
+                    "default": 0.0
+                },
+                "reveal_answer": {
+                    "type": "boolean",
+                    "description": "Whether or not to reveal the answer when the question times out. Defaults to true.",
+                    "default": true
+                },
+                "timeout": {
+                    "type": ["integer", "number"],
+                    "description": "Timeout per question in seconds, must be positive number greater than 0.0. Defaults to 120.0.",
+                    "default": 120.0
+                },
+                "use_spoilers": {
+                    "type": "boolean",
+                    "description": "Whether to use spoiler markdown when revealing question answers. Defaults to false.",
+                    "default": false
+                }
+            }
```

The only valid keys are specified in the schema so we can disallow names not listed here:
```suggestion
            },
            "additionalProperties": false
```